RAGchain/utils/linker/base.py

```diff
@@ -0,0 +1,17 @@
+from abc import ABC, abstractmethod
+from typing import Union
+from uuid import UUID
+
+
+class BaseLinker(ABC):
+    @abstractmethod
```

Comment:
use metaclass for making BaseLinker to singleton.
https://alphahackerhan.tistory.com/34

Reply:
Thx to your kindness blog link.